org.jacoco.ant.test/src/org/jacoco/ant/SecurityManagerTest.java

```diff
@@ -0,0 +1,40 @@
+/*******************************************************************************
+ * Copyright (c) 2009, 2024 Mountainminds GmbH & Co. KG and Contributors
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Brock Janiczak - initial API and implementation
+ *
+ *******************************************************************************/
+package org.jacoco.ant;
+
+import java.io.File;
+
+import org.apache.ant.antunit.junit3.AntUnitSuite;
+import org.apache.ant.antunit.junit4.AntUnitSuiteRunner;
+import org.jacoco.core.test.validation.JavaVersion;
+import org.junit.runner.RunWith;
+
+import junit.framework.TestSuite;
+
+@RunWith(AntUnitSuiteRunner.class)
+public class SecurityManagerTest {
+
+	public static TestSuite suite() {
+		if (JavaVersion.current().isBefore("24")) {
+			System.setProperty("org.jacoco.ant.securityManagerTest.classes.dir",
+					TestTarget.getClassPath());
+			final File file = new File(
+					"src/org/jacoco/ant/SecurityManagerTest.xml");
+			return new AntUnitSuite(file, SecurityManagerTest.class);
+		}
+		// No SecurityManager any more in Java 24
+		final File file = new File("src/org/jacoco/ant/empty.xml");
```

Comment:
@marchof I guess that `AntUnitSuiteRunner` does not understand `org.junit.AssumptionViolatedException` that we use in other places and that's why we need `empty.xml` here?

Reply:
No, this results in an initialization exception. Also I tried to return an empty `TestSuite` but the `AntUnitSuiteRunner` expects a `AntUnitSuite` instance.